src/components/Starred.jsx

```diff
@@ -0,0 +1,42 @@
+import { useSelector, useDispatch } from 'react-redux'
+import { Link } from 'react-router-dom'
+import starredSlice from '../data/starredSlice'
+import Movie from './Movie'
+import '../styles/starred.scss'
+
+const Starred = ({viewTrailer}) => {
```

Comment:
- To keep our code clean and maintainable, let's follow the Single Responsibility Principle (SRP) by separating the movie listing and footer logic into separate components.
- Create a StarredMovieList.jsx component to handle the movie listing.
- Create a StarredFooter.jsx component to handle the footer logic.
- Keep the Starred component as a container that organizes these components.